packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx

```diff
@@ -118,6 +136,29 @@ export const CloudPulseResourcesSelect = React.memo(
         placeholder={
           selectedResources?.length ? '' : placeholder || 'Select Resources'
         }
+        renderOption={(props, option) => {
+          const { key, ...rest } = props;
+          const isResourceSelected = selectedResources?.some(
+            (item) => item.label === option.label
+          );
+          const isOverLimitOption =
+            selectedResources &&
+            selectedResources?.length >= setResourceLimit &&
```

Comment:
Replace "selectedResources?.length" with "selectedResources.length" since you already checked for  selectedResources.

Reply:
this & above all comments fixed and pushed..